### ISSUE TITLE
Cant add spotify account

### ISSUE THREAD

**Issue by TrickShotMLG02** [Birk3n/TS3Spotify#14]
I load the plugin and add an account in pm of the bot. It says "Lets check it quickly."

In the console these lines appear
![grafik](https://user-images.githubusercontent.com/43967908/166139957-cd397602-038a-491a-b4bf-e4378390b34c.png)

Also on my spotify account i have a new device, where i can stream music to.

But the bot doesn't play any music and tells me everytime, that i dont have an account setted up.

**Comment by TrickShotMLG02** [comment]
The log from the bot:
```
10:22:29.1294| INFO|Core| [============ TS3AudioBot started =============]
10:22:29.1603| INFO|Core| [ Date: Sunday, 01 May 2022 10:22:29
10:22:29.1859| INFO|Core| [ Version: 0.12.3--no-branch-.1/(no branch)/4c6be036
10:22:29.1859| INFO|Core| [ Build: Release
10:22:29.1859| INFO|Core| [ Platform: Ubuntu 20.04.4 LTS (Focal Fossa) (64bit)
10:22:29.1877| INFO|Core| [ Runtime: .NET Core (3.1.24) ServerGC:False GC:SustainedLowLatency
10:22:29.1877| INFO|Core| [ Opus: libopus 1.3.1 (x64)
10:22:29.1877| INFO|Core| [==============================================]
10:22:29.5170|ERROR|Core| Can't find a WebInterface path to host. Try specifying the path to host in the config
10:22:29.5170| INFO|Core| Starting Webserver on port 58913
10:22:29.6633| INFO|Core| Bot "default" connecting to "192.168.1.4"
10:22:30.0584| INFO|Core| Client connected.
10:22:44.9734| INFO|0| User TrickShotMLG requested: !plugins load 0
10:22:45.0525| INFO|0| * Spotify initialized
10:22:45.0878| INFO|0| * Config loaded
10:22:49.7762| INFO|0| User TrickShotMLG requested: !spoitfy account
10:22:51.4924| INFO|0| User TrickShotMLG requested: !spoitfy account add
Starting with  -b 160 --enable-volume-normalisation --initial-volume 100 --backend pipe -n TS3AudioBot -c "/home/tim/librespot/target/release" --device /tmp/TS3AudioBotFifo -u MYEMAILADDRESS -p MYPASSWORD
10:22:59.7066|ERROR|| Unobserved Task error!
10:22:59.7193|ERROR|| Unobserved Task error!
10:24:56.4650| INFO|0| User TrickShotMLG requested: !spoitfy account
10:25:00.8745| INFO|0| User TrickShotMLG requested: !plugins unload 0
rm: cannot remove 'TS3AudioBotFifo': No such file or directory
10:25:04.6108| INFO|0| User TrickShotMLG requested: !plugins load 0
```


The bot.toml:

```
#Starts the instance when the TS3AudioBot is launched.
run = true

[commands]

[commands.alias]

[connect]
#The server password. Leave empty for none.
server_password = { pw = "" }
#The default channel password. Leave empty for none.
channel_password = {  }
#Overrides the displayed version for the ts3 client. Leave empty for default.
client_version = {  }
#The address, ip or nickname (and port; default: 9987) of the TeamSpeak3 server
address = "192.168.1.4"

[connect.identity]
#||| DO NOT MAKE THIS KEY PUBLIC ||| The client identity. You can import a teamspeak3 identity here too.
key = "BOT_ID_KEY"
#The client identity offset determining the security level.
offset = 1734

[reconnect]

[audio]
#When a new song starts the volume will be trimmed to between min and max.
#When the current volume already is between min and max nothing will happen.
#To completely or partially disable this feature, set min to 0 and/or max to 100.
volume = {  }

[playlists]

[history]

[events]

[spotify]
deviceName = "TS3AudioBot"
pipelineName = "TS3AudioBotFifo"
librespotPath = "/home/tim/librespot/target/release/librespot"

[[spotify.accountList]]
id = ""
email = "MYEMAILADDRESS"
username = ""

```

Is it correct that id and username are empty in the bot.toml file?

**Comment by Birk3n** [comment]
First, I think the id in bot.toml is necessary but the username not anymore and this could be an error after logged in.

Second, I think you dont hear something because there is a problem with the Fifo /tmp/TS3AudioBotFifo. Can you run `mkfifo /tmp/TS3AudioBotFifo` with the TS3Bot user?

**Comment by TrickShotMLG02** [comment]
the log after starting the bot
![image](https://user-images.githubusercontent.com/43967908/166160260-10257e29-cd86-41a6-b0e9-8a6e9073b0b8.png)

![image](https://user-images.githubusercontent.com/43967908/166160315-5c0ef8ed-497e-4d79-89d4-61621386ea8c.png)
and the result of the command

**Comment by TrickShotMLG02** [comment]
I got it to work after stopping the bot and entering my uid into the id in the toml

**Comment by TrickShotMLG02** [comment]
Fixed it by moving the credentials.json file to the directory of my current ts3 uid

This is the new log of my ts3audiobot console
```
09:07:20.3125| INFO|Core| [============ TS3AudioBot started =============]
09:07:20.3379| INFO|Core| [ Date: Monday, 02 May 2022 09:07:20
09:07:20.3587| INFO|Core| [ Version: 0.12.3--no-branch-.1/(no branch)/4c6be036
09:07:20.3587| INFO|Core| [ Build: Release
09:07:20.3587| INFO|Core| [ Platform: Ubuntu 20.04.4 LTS (Focal Fossa) (64bit)
09:07:20.3587| INFO|Core| [ Runtime: .NET Core (3.1.24) ServerGC:False GC:SustainedLowLatency
09:07:20.3587| INFO|Core| [ Opus: libopus 1.3.1 (x64)
09:07:20.3610| INFO|Core| [==============================================]
09:07:20.6607|ERROR|Core| Can't find a WebInterface path to host. Try specifying the path to host in the config
09:07:20.6617| INFO|Core| Starting Webserver on port 58913
09:07:20.8026| INFO|Core| Bot "default" connecting to "192.168.1.4"
09:07:21.1670| INFO|Core| Client connected.
09:07:28.4434| INFO|0| User TrickShotMLG requested: !plugins load 0
09:07:28.5120| INFO|0| * Spotify initialized
09:07:28.5463| INFO|0| * Config loaded
09:07:31.6524| INFO|0| User TrickShotMLG requested: !spotfy play
Starting with  -b 160 --enable-volume-normalisation --initial-volume 100 --backend pipe -n TS3AudioBot -c "/home/tim/librespot/target/release/**MY_UID**" --device /tmp/TS3AudioBotFifo
```

**Comment by TrickShotMLG02** [comment]
To sum it up

- download and build librespot
- Clone the current Repository as given in the readme
- Build the bot, move the spotify plugin into the plugins folder
- Create folder lib in pluginsfolder
- Move NAudio.Core.dll and NAudio.dll into the lib folder
- start the bot with `dotnet TS3Audiobot.dll`
- Load the plugin with `!plugins load <id>`
- Run `!spotify account add` and follow the instructions
- stop the whole audiobot by terminating the process
- go to the bots/default directory and edit the bot.toml
- At the bottom you will see something like [[spotify.accountList]] with your email-address
- Also enter the absolute path of librespot under [spotify]
- Enter your uid from teamspeak into the id (you can find your unique id from your current identity in the identity window (ctrl+i))
- save the toml file and start the bot again with `dotnet TS3Audiobot.dll`
- Load the plugin with `!plugins load <id>`
- Run `!spotify account` and you should see your email address
- Now you can run the command `!spotify start`
- in the console you should see a line like this `Starting with  -b 160 --enable-volume-normalisation --initial-volume 100 --backend pipe -n TS3AudioBot -c "**YOUR_LIBRESPOT_PATH**/**YOUR_UID**" --device /tmp/TS3AudioBotFifo`
- now go to the librespot path and move the credentials.json into the directory named by your UID.
- You may need to restart the audiobot once again and it should work.
- To stream music, open spotify, select the TS3AudioBot as Device and play your music

**Comment by Birk3n** [comment]
Fixed it in a979847fd9796a0f4e1a530fec7a57c6c9043aab
